src/stories/Autocomplete/Hook/index.tsx

```diff
@@ -119,32 +119,38 @@ export function HooksTemplate(args) {
             if (!section?.data?.length) {
               return null;
             }
+
             const { type, displayName } = section;
-            let sectionName = section.displayName;
+            let sectionTitle: string;
 
             switch (type) {
               case 'recommendations':
-                sectionName = section.podId;
+                sectionTitle = section.podId;
                 break;
               case 'custom':
-                sectionName = toKebabCase(displayName);
+                sectionTitle = displayName;
                 break;
               case 'autocomplete':
-                sectionName = section.indexSectionName;
+                sectionTitle = section.indexSectionName;
                 break;
               default:
-                sectionName = section.indexSectionName;
+                sectionTitle = section.indexSectionName;
                 break;
```

Comment:
This is (type) broken currently but we still need it. Sections can be one of type `autocomplete` | `recommendations` | `custom`, but the type on `autocomplete` is optional meaning it can be `undefined`.

There's two ways to fix this:
 1. Define an additional type that mimics `AutocompleteConfiguration` but with no type
 2. Empty the default clause here. Add an if-conditional above to check if `type===undefined`
 3. Assign the type during `convertLegacyParametersAndAddDefaults` in `useCioAutocomplete` (preferred)

I'll pull up a separate PR for this.

Reply:
which part you mean "This is (type) broken currently"? @Mudaafi

Reply:
@mocca102 Might be my configuration, but since we've explicitly defined the type `Section` to be one of three things and the switch statement accounts for all three, the final default type is `never`. TS then throws the error `Property 'indexSectionName' does not exist on type 'never'`